cmds/ldd/ldd.go

```diff
@@ -12,55 +12,35 @@
 package main
 
 import (
-	"debug/elf"
-	"errors"
 	"fmt"
+	"log"
 	"os"
-)
 
-var list map[string]bool
+	"github.com/u-root/u-root/pkg/ldd"
+)
 
-func process(file *os.File, name string) error {
-	if f, err := elf.NewFile(file); err != nil {
-		return err
-	} else {
-		if fl, err := f.ImportedLibraries(); err != nil {
-			return err
-		} else {
-			if s := f.Section(".interp"); s == nil {
-				return errors.New("No interpreter")
-			} else {
-				if interp, err := s.Data(); err != nil {
-					return err
-				} else {
-					// We could just append the interp but people
-					// expect to see that first.
-					fl = append([]string{string(interp)}, fl...)
-					for _, i := range fl {
-						list[i] = true
-					}
-				}
+func usage() {
+	log.Fatalf("usage: ldd file [file...]")
+}
 
-			}
+func ldd(s ...string) ([]string, error) {
+	var libs []string
+	l, err := uroot.Ldd(s)
+	if err != nil {
+		return nil, err
+	}
+	for _, i := range l {
+		if i.Mode().IsRegular() {
+			libs = append(libs, i.FullName)
 		}
 	}
-	return nil
+	return libs, nil
 }
 
 func main() {
-	list = make(map[string]bool)
-	if len(os.Args) < 2 {
-		process(os.Stdin, "stdin")
```

Comment:
it did not seem worth it.
In principle, 
ldd /dev/fd/0 < /bin/date
ought to work. Evidently not on Unix. Too bad.